src/drivers/actuators/vertiq_io/vertiq_client_manager.hpp

```diff
@@ -0,0 +1,275 @@
+	* @brief Gives access to the object ID currently being used
+	*
+	* @return The value stored in _object_id_now
+	*/
+	uint8_t GetObjectIdNow();
+
+	/**
+	* @brief When the target module ID parameter changes, we need to delete and remake all of our clients in order to make sure that we're talking to the
+	*        correct motor.
+	*
+	* @param new_object_id The new target module ID that we should use to instantiate our new clients
+	*/
+	void UpdateClientsToNewObjId(uint8_t new_object_id);
+
+	/**
+	* @brief Adds a new client to our array of Configuration Clients. Configurations clients are those meant to interface a configurable module parameter
+	* with a PX4 parameter. These are clients whose module ID will change over time, and will be dynamically updated
+	*
+	* @param client a pointer to the new client
+	*/
+	void AddNewConfigurationClient(ClientAbstract * client);
+
+	/**
+	* @brief Adds a new client to our array of Operational Clients. Operational clients are those meant to hold an operational client such as those used
+	* for direct motor control. Operational clients should have a constant module ID, and should be made only once
+	*/
+	void AddNewOperationalClient(ClientAbstract * client);
+
+	/**
+	* @brief Returns the number of clients added to our Configuration Clients array
+	*
+	* @return The value _configuration_clients_in_use
+	*/
+	uint8_t GetNumberOfConfigurationClients();
+
+	/**
+	* @brief Returns the number of clients added to our Operational Clients array
+	*
+	* @return The value _operational_clients_in_use
+	*/
+	uint8_t GetNumberOfOperationalClients();
+
+	/**
+	* @brief Returns a pointer to our IFCI client used for telemetry
+	*
+	* @return The _telem_ifci pointer
+	*/
+	IQUartFlightControllerInterfaceClient *GetTelemIFCI();
+
+	/**
+	* @brief Creates and adds a new entry wrapper object to our array of entry wrappers
+	*
+	* @param px4_param A parameter stored in PX4. This can be found with the param_find function
+	* @param entry A pointer to a Vertiq client entry
+	*/
+	template <typename iquart_data_type , typename px4_data_type>
+	void AddNewClientEntry(param_t px4_param, ClientEntryAbstract *entry){
+		if(_added_entry_wrappers < MAX_CLIENT_ENTRIES){
+			_entry_wrappers[_added_entry_wrappers] = new EntryWrapper<iquart_data_type, px4_data_type>;
+			_entry_wrappers[_added_entry_wrappers]->ConfigureStruct(px4_param, entry);
+			_added_entry_wrappers++;
+		}else{
+			PX4_INFO("Could not add this entry. Maximum number exceeded");
+		}
+	}
+
+private:
+	/**
+	* @brief Initialize all of the Vertiq Clients that we want to use
+	*
+	* @param object_id The object ID with which to initialize our clients
+	*/
+	void InitVertiqClients(uint8_t object_id);
+
+	/**
+	* @brief Initialize all of the Entry Wrappers
+	*/
+	void InitEntryWrappers();
+
+	uint8_t _object_id_now;
+
+	//We need a serial handler in order to talk over the serial port
+	VertiqSerialInterface *_serial_interface;
+
+	//These exist whenever we're using IQUART
```

Comment:
Can you expand on the commenting here to re-iterate what the point of making all these entries is for people poking through the code. Why do I need both a bunch of clients and a bunch of entries? (I know the answer, I just want it to be obvious to someone when they look here)